shpkpr/deployment/bluegreen/deployment.py

```diff
@@ -143,7 +145,19 @@ def get_deployment_group(app_definition):
             return labels.get('HAPROXY_DEPLOYMENT_GROUP')
 
         target_deployment_group = get_deployment_group(self.app_definition)
-        for app_definition in self.marathon_client.list_applications():
+        for app_definition in self._fetch_apps_state:
             if get_deployment_group(app_definition) == target_deployment_group:
                 return app_definition
         return None
+
+    @cached_property
+    def _fetch_marathon_info(self):
```

Comment:
`_fetch` seems unusual for naming of a property, since you're not _really_ calling it, just treating it like a variable, same for the one below. NAB though, can be fixed later.

Reply:
good point, something i learned through the process was that these aren't functions anymore :) 

I'll merge this (lets talk about versioning and I can test with one of my services), then I'll put a PR to change this